tests/ui/from_instead_of_into.rs

```diff
@@ -0,0 +1,29 @@
+// run-rustfix
+#![allow(unused_imports)]
+#![warn(clippy::from_instead_of_into)]
+use std::convert::TryFrom;
+
+fn foo<T>(a: T)
+where
+    u32: From<T>,
+{
+}
+
+fn foo1<T>(a: T)
+where u32: Copy + Clone + From<T>,
+{
+}
+
+fn bar<T>(a: T)
+where
+    u32: TryFrom<T>,
+{
+}
+
+fn bar1<T>(a: T)
+where
+    u32: Copy + TryFrom<T> + Clone,
```

Comment:
Doubts: I have added two multiply generic bound predicate here, do I need more predicate?

Reply:
How should I add more generic bounds?

Lengthen the origin bounds and put `From<T>` at the first, middle, and last should be enough?